frontend/src/services/logic/RoomFilter.js

```diff
@@ -27,6 +27,14 @@ class RoomFilter {
          if ( !roomName.includes( filterName ) ) return false;
       }
 
+      if ( filters.building && filters.building.length > 0 ) {
+         if ( !room.location ) return false;
+         const location = `${room.location.building}_${room.location.floorNo}`;
+         const selectedThisFloor = filters.building.includes( location );
+         const selectedThisBuilding = filters.building.includes( room.location.building );
+         if ( !( selectedThisFloor || selectedThisBuilding ) ) return false;
+      }
+
       return true;
    }
```

Comment:
It would be easier to maintain if filtering was divided into multiple functions. Then you have functions for `location`, `number of seats` etc. and general filter function like this 
```js
function filterRooms(rooms, filters) {
  return rooms.filter(room => filterFn1(room, filters) && filterFn2(room, filters) && ...);
}
```  
where you can just add more filter functions.

Reply:
Fixed in #77 .